### DISCUSSION TITLE
Is this package suitable for syncing liteDB?

### DISCUSSION THREAD

**Discussion by herczegzoltan** [CommunityToolkit/Datasync#170] [Q&A] [answered]
Hello,

I’m unsure if this package is the right solution for my use case and would appreciate some guidance.

Here’s my architecture:

Client: Maui

Uses LiteDB to store data locally.

Server: ASP.NET Core

Uses Identity for user registration.

MongoDB for storing data pushed from clients.

When I save data locally, I mark it as "Changed" and trigger a push to the server. The push includes a token to authorize the logged in user. So on the server side before saving, the user Id is attached to the data so can be saved to MongoDB with an associated ownerId. To achieve this, I’ve implemented a custom sync mechanism. However, I’m looking for a solution that can handle syncing locally stored data to the server automatically when the internet is available.

Can this package, or any other package, replace my current sync implementation? I haven’t found any examples using LiteDB as the local store.

Any information would be greatly appreciated!
Thanks!

**Comment by adrianhall** [comment, marked as answer]
I doubt you will find any library that uses LiteDB as a local store.  Technically, there is no reason why LiteDB can't be used.  However, all the solutions out there either use EF (and LiteDB does not support EF) or they use their own store.
